### PR TITLE
cpu training + wandb logging

### PR DESCRIPTION
**Pull request by KSGulin**
Added option to run without CUDA
Added WandB logging

### CODE REVIEW

File: train.py [hunk 1]
```diff
@@ -318,28 +320,32 @@ def main():
     
     if args.log_wandb:
         if has_wandb:
-            wandb.init(project=args.experiment, config=args)
+            wandb.init(project=args.experiment, config=args, dir = './wandb_logs')
```
Comment (bfineran) [inline train.py]: can we set the `dir` to a subdirectory the main log directory

**Review by bfineran — COMMENTED** [review]
looks good overall, just a comment to see where local wandb files get logged to

File: train.py (hunk 1 above)
Comment (KSGulin) [reply train.py]: Good call, updated to write to same directory as all output files.

Comment (bfineran) [reply train.py]: great, let's still keep it under a `wandb` subdirectory of the output directory

Comment (KSGulin) [reply train.py]: It automatically creates a `wandb`subdirectory in the target dir

**Review by bfineran — APPROVED** [review]

**Comment by bfineran** [comment]
LGTM. @KSGulin let's get the PR title updated to reflect that this adds support for CPU training and WandB logging

**Review by rahul-tuli — APPROVED** [review]

**Pull request merged**